pubsub/pubsub.go

```diff
@@ -0,0 +1,52 @@
+package pubsub
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+)
+
+// Publisher defines the sender contract of a basic publish/subscribe design pattern.
+type Publisher struct {
+	sync.Mutex
+	subChannels []chan *Event
+}
+
+type Event struct {
+	Context context.Context
+	Payload interface{}
+}
+
+func NewPublisher() *Publisher {
+	return &Publisher{
+		subChannels: []chan *Event{},
+	}
+}
+
+// A single Publisher instance is able to take several subscribers.
+// It returns a receive-only channel.
+func (pub *Publisher) Subscribe() <-chan *Event {
+	ch := make(chan *Event)
```

Comment:
This is an unbuffered channel, might want to allow the subscriber to choose the amount of buffer via an arg to `Subscribe`